metrics/node/subscribe.go

```diff
@@ -0,0 +1,47 @@
+package node
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/status-im/status-go/geth/log"
+)
+
+// SubscribeServerEvents subscribes to server and listens to
+// PeerEventTypeAdd and PeerEventTypeDrop events.
+func SubscribeServerEvents(ctx context.Context, node *node.Node) error {
+	server := node.Server()
+	if server == nil {
+		return errors.New("server is unavailable")
+	}
+
+	ch := make(chan *p2p.PeerEvent)
```

Comment:
my comment got lost somehow, so there is a suggestion on Feed.Subscribe method

> // The channel should have ample buffer space to avoid blocking other subscribers.
// Slow subscribers are not dropped.

Reply:
Our subscribers are very simple so not sure if it does matter. Also, they are only used if status-go is built with `metrics` tag. As we skip any event not related to adding/removing a peer, we may bound the buffer size to the max peers attribute.

Reply:
It's done.